sklearn/preprocessing/_data.py

```diff
@@ -1336,8 +1339,10 @@ def maxabs_scale(X, *, axis=0, copy=True):
         otherwise (if 1) scale each sample.
 
     copy : bool, default=True
-        Set to False to perform inplace scaling and avoid a copy (if the input
-        is already a numpy array).
+        If False, try to avoid a copy and scale in place.
+        This is not guaranteed to always work in place; e.g. if the data is
+        not a NumPy array, or is a scipy.sparse CSR matrix, or is not of dtype
+        float, a copy may still be returned.
```

Comment:
`check_array(..., accept_sparse=("csr", "csc"), dtype=FLOAT_DTYPES)` means there is no copy if the matrix is a CSC or CSR matrix and the dtype is of kind float